internal/store/pod_test.go

```diff
@@ -1495,8 +1498,8 @@ func TestPodStore(t *testing.T) {
 	}
 
 	for i, c := range cases {
-		c.Func = generator.ComposeMetricGenFuncs(podMetricFamilies(nil))
-		c.Headers = generator.ExtractMetricFamilyHeaders(podMetricFamilies(nil))
+		c.Func = generator.ComposeMetricGenFuncs(podMetricFamilies(c.AllowLabelsList))
```

Comment:
This is great but ideally, we want a separate test case for this. Keep testing what we did before and add this as a new test.

Reply:
Agreed, updated